internal/cluster/statemachine.go

```diff
@@ -183,13 +185,56 @@ func NewClusterStateMachine(th *transitionHandler) stateswitch.StateMachine {
 		PostTransition:   th.PostRefreshCluster(statusInfoPreparingForInstallationTimeout),
 	})
 
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefreshStatus,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.ClusterStatusInstallingPendingUserAction),
+		},
+		Condition:        th.HasHostsInError,
```

Comment:
This condition is not correct. we will fail the installation only if one of the masters is in error 
Take a look at `IsInstalling` condition.